src/parsers/basic_expression.cr

```diff
@@ -33,7 +33,8 @@ module Mint
         parallel ||
         try_expression ||
         case_expression ||
-        parenthesized_expression_or_inline_function ||
+        parenthesized_expression_no_error ||
```

Comment:
It seems that the error produced by `parenthesized_expression` is only checked in the specs. But the actual parser just swallowed those errors and tried to parse an inline function. So now there's an unused `parenthesized_expression` (well, still used in specs) so I'm not sure this is an ideal state. Let me know!

Reply:
I guess it changed at some point, so If it's not used in the parser itself, we can remove it and have only one `parenthesized_expression` which returns ˙Ast::ParenthesizedExpression?`

Reply:
Done! This also removed some error types that apparently were unused.